archaius2-core/src/main/java/com/netflix/archaius/DefaultAppConfig.java

```diff
@@ -205,46 +182,40 @@ public static Builder builder() {
         return new Builder();
     }
     
-    public static DefaultAppConfig createDefault() {
+    public static DefaultAppConfig createDefault() throws ConfigException {
         return new Builder().build();
     }
     
+    private final CascadingCompositeConfig     root;
     private final SimpleDynamicConfig          runtime;
-    private final CascadingCompositeConfig     override;
+    private final CascadingCompositeConfig     dynamic;
     private final CascadingCompositeConfig     library;
     private final DefaultConfigLoader          loader;
     private final PropertyFactoryDynamicConfigListener dynamicObserver;
     
-    public DefaultAppConfig(Builder builder) {
-        super(NAME);
+    public DefaultAppConfig(Builder builder) throws ConfigException {
+        super("APP_CONFIG");
         
         try {
+            this.root = new CascadingCompositeConfig(NAME);
+            
             if (builder.decoder != null) {
                 setDecoder(builder.decoder);
+                this.root.setDecoder(builder.decoder);
             }
 
             this.setStrInterpolator(builder.interpolator.create(this));
+            this.root.setStrInterpolator(this.getStrInterpolator());
 
             // The following are added first, before application configuration
             // to allow for replacements in the application configuration cascade
             // loading.
-            super.addConfig(runtime = new SimpleDynamicConfig(OVERRIDE_LAYER));
+            root.addConfig(runtime = new SimpleDynamicConfig(OVERRIDE_LAYER));
             runtime.setProperties(builder.props);
             
-            if (builder.enableOverrideLayer) {
-                super.addConfig(override = new CascadingCompositeConfig(DYNAMIC_LAYER));
-            }
-            else {
-                override = null;
-            }
-            
-            if (builder.enableSystemLayer) {
-                super.addConfig(new SystemConfig());
-            }
-            
-            if (builder.enableEnvironmentLayer) {
-                super.addConfig(new EnvironmentConfig());
-            }
+            root.addConfig(dynamic = new CascadingCompositeConfig(DYNAMIC_LAYER));
+            root.addConfig(new SystemConfig());
+            root.addConfig(new EnvironmentConfig());
```

Comment:
What is the rationale for not providing the option of leaving out system properties and environment variables? For test cases especially I would prefer to leave these out so I know my test case is isolated.